drivers/sensor/bmp581/bmp581.c

```diff
@@ -0,0 +1,565 @@
+
+		return BMP5_OK;
+	}
+	case SENSOR_CHAN_PRESS:
+		/* returns pressure in Pa */
+		*val = drv->last_sample.pressure;
+		return BMP5_OK;
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		/* returns temperature in Celcius */
+		*val = drv->last_sample.temperature;
+		return BMP5_OK;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static int set_iir_config(const struct sensor_value *iir, struct bmp581_data *drv)
+{
+	int ret = BMP5_OK;
+
+	if (iir != NULL) {
+		/* Variable to store existing powermode */
+		enum bmp5_powermode prev_powermode;
+
+		ret = get_power_mode(&prev_powermode, drv);
+		if (ret != BMP5_OK) {
+			LOG_DBG("Not able to get current power mode.");
+			return ret;
+		}
+		/* IIR configuration is writable only during STANDBY mode(as per datasheet) */
+		set_power_mode(BMP5_POWERMODE_STANDBY, drv);
+
+		/* update IIR config */
+		uint8_t dsp_config[2];
+
+		ret = reg_read(BMP5_REG_DSP_CONFIG, dsp_config, 2, drv);
+		if (ret != BMP5_OK) {
+			LOG_DBG("Failed to read dsp config register.");
+			return ret;
+		}
+		/* Put IIR filtered values in data registers */
+		dsp_config[0] =
+			BMP5_SET_BITSLICE(dsp_config[0], BMP5_SHDW_SET_IIR_TEMP, BMP5_ENABLE);
+		dsp_config[0] =
+			BMP5_SET_BITSLICE(dsp_config[0], BMP5_SHDW_SET_IIR_PRESS, BMP5_ENABLE);
+
+		/* Configure IIR filter */
+		dsp_config[1] = iir->val1;
+		dsp_config[1] = BMP5_SET_BITSLICE(dsp_config[1], BMP5_SET_IIR_PRESS, iir->val2);
+
+		/* Set IIR configuration */
+		ret = reg_write(BMP5_REG_DSP_CONFIG, dsp_config, 2, drv);
+
+		if (ret != BMP5_OK) {
+			LOG_DBG("Failed to configure IIR filter.");
+			return ret;
+		}
+
+		/* Restore previous power mode if it is not standby already */
+		if (prev_powermode != BMP5_POWERMODE_STANDBY) {
+			ret = set_power_mode(prev_powermode, drv);
+		}
+	} else {
+		ret = BMP5_E_NULL_PTR;
+	}
+
+	return ret;
+}
+
+static int bmp581_attr_set(const struct device *dev, enum sensor_channel chan,
+			   enum sensor_attribute attr, const struct sensor_value *val)
+{
+	struct bmp581_data *drv = (struct bmp581_data *)dev->data;
+	int ret = -ENOTSUP;
+
+	switch ((int)attr) {
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		ret = set_odr_config(val, drv);
+		break;
+	case SENSOR_ATTR_OVERSAMPLING:
+		ret = set_osr_config(val, chan, drv);
+		break;
+	case BMP5_ATTR_POWER_MODE:
+	{
+		enum bmp5_powermode powermode = (enum bmp5_powermode)val->val1;
+
+		ret = set_power_mode(powermode, drv);
+		break;
+	}
+	case BMP5_ATTR_IIR_CONFIG:
+		ret = set_iir_config(val, drv);
+		break;
+	default:
+		ret = -ENOTSUP;
+		break;
+	}
+	return ret;
+}
+
+static int bmp581_init(const struct device *dev)
+{
+	const struct bmp581_config *cfg = (struct bmp581_config *)dev->config;
+	struct bmp581_data *drv = (struct bmp581_data *)dev->data;
+	int ret = -1;
+
+	/* Reset the chip id. */
+	drv->chip_id = 0;
+	memset(&drv->osr_odr_press_config, 0, sizeof(drv->osr_odr_press_config));
+	memset(&drv->last_sample, 0, sizeof(drv->last_sample));
+
+	drv->i2c = cfg->i2c;
+
+	drv->i2c_addr = cfg->i2c_addr;
```

Comment:
This shouldn't be needed

Reply:
removed `struct i2c_dt_spec i2c` from `struct bmp581_data`